hash/structhash/structhash_test.go

```diff
@@ -69,6 +69,37 @@ func TestDump(t *testing.T) {
 			}{nil},
 			"{a:nil}",
 		},
+		{
+			struct {
+				a interface{} `hash:"name:a"`
+			}{
+				struct {
+					b map[string]int `hash:"name:b"`
+				}{
+					map[string]int{
+						"c": 1,
+					},
+				},
+			},
+			"{a:{b:(\"c\":1)}}",
+		},
+		// NOTE: structhash will allow to process all interface types.
+		// gogo/protobuf is not able to set tags for directly oneof interface.
+		// see: https://github.com/gogo/protobuf/issues/623
+		{
```

Comment:
Another, and way worst problem. The following is returning `{a:{b:1}}`:
```
		},
		{
			struct {
				a interface{}
			}{
				struct {
					b interface{}
				}{
					b: 1,
				},
			},
			"{}",
		},
```

Reply:
Solution, replace in [structhash.go line 105](https://github.com/mesg-foundation/engine/pull/1318/files#diff-dfaac64e3ba689ae3bd308fca77f695cR105):
```diff
- if to.name == "" && reflect.Zero(sf.Type).Kind() == reflect.Interface {
+ if to.name == "" && v.CanInterface() {
```

Reply:
Solution, replace in [structhash.go line 105](https://github.com/mesg-foundation/engine/pull/1318/files#diff-dfaac64e3ba689ae3bd308fca77f695cR105) by:
```diff
- if to.name == "" && reflect.Zero(sf.Type).Kind() == reflect.Interface {
+ if to.name == "" && v.Field(i).Kind() == reflect.Interface && v.Field(i).Elem().Kind() == reflect.Struct {
```

Reply:
This is how it interface works. It's not a bug, it done design.
So let's break it down:

a `struct` is passed with `a interface{]` fileds which has no tag.

A struct is serialized and `a` field is serialized because it's an interface.

Then `interface{}` has struct inside. That struct is serialized because it's not a field of any struct, but rather the object (interface value). The struct contains `b` fields which is an interface so it's serialized and  interface value contains an integer. So the final output should be `{a:{b:1}}`